shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1396,12 +1508,32 @@ - (void)setupKeyboardAnimationVsyncClient {
       // Ensure the keyboardAnimationView is in view hierarchy when animation running.
       [flutterViewController.get().view addSubview:[flutterViewController keyboardAnimationView]];
     }
-    if ([flutterViewController keyboardAnimationView].layer.presentationLayer) {
-      CGFloat value =
-          [flutterViewController keyboardAnimationView].layer.presentationLayer.frame.origin.y;
-      flutterViewController.get()->_viewportMetrics.physical_view_inset_bottom = value;
-      [flutterViewController updateViewportMetrics];
-    }
+
+    KeyboardAnimationView* keyboardView =
+        (KeyboardAnimationView*)[flutterViewController keyboardAnimationView];
+
+    fml::TimeDelta timeElapsed = recorder.get()->GetVsyncTargetTime() - keyboardView.startTime;
+    double maxFrameRate = [DisplayLinkManager displayRefreshRate];
+    double frameRate = round(
+        1 /
+        (recorder.get()->GetVsyncTargetTime() - recorder.get()->GetVsyncStartTime()).ToSecondsF());
+
+    double animationSettlingDuration = [flutterViewController keyboardSpringCurve].settlingDuration;
+    double keyboardAnimationTimeMillis = animationSettlingDuration * 1000;
+
+    double expectedFrames = frameRate * animationSettlingDuration * (maxFrameRate / frameRate);
+    double percentComplete = timeElapsed.ToMillisecondsF() / keyboardAnimationTimeMillis;
+
+    int frameApproximation = MIN(roundf(MIN(percentComplete, 1) * expectedFrames),
+                                 [[flutterViewController keyboardAnimationStops] count]);
+    double animationFramePercentage = [[[flutterViewController keyboardAnimationStops]
+        objectAtIndex:frameApproximation] doubleValue];
```

Comment:
1. I think using 
```objc
double frameRate = [[flutterViewController keyboardAnimationVSyncClient] getRefreshRate];
``` 
is ok, no need for recalculate again using `FrameTimingsRecorder`

2. Nits for `keyboardAnimationStops`'s count, using dot syntax, it is a property in Objc instead of a method.
3. Max index should be `[flutterViewController keyboardAnimationStops] count] - 1`

Reply:
:) slowly learning more Objc